### ISSUE TITLE
LDFLAGS are ignored when configuring MoarVM

### ISSUE THREAD

**Issue by mojca** [MoarVM/MoarVM#472]
When running:
```sh
env LDFLAGS=-L/opt/local/lib /opt/local/bin/perl Configure.pl --prefix=/opt/local --has-dyncall
```
I get
```
Welcome to MoarVM!

Configuring native build environment ................... OK
    trying to compile a simple C program ............... ERROR

    Can't link simple C program.
    Failing command: clang -o try try.o -ldyncall_s -ldyncallback_s -ldynload_s -lpthread 2>&1
    Error: ld: library not found for -ldyncall_s
clang: error: linker command failed with exit code 1 (use -v to see invocation)


Cannot continue after this error.
```

The problem seems to be that `LDFLAGS=-L/opt/local/lib` is not respected and the linker fails to find the libraries. ~~Under some circumstances this works, but it's not clear to me when exactly~~. The reason why it works on newer platforms is because the package manager sets `LIBRARY_PATH=/opt/local/lib`. That is respected with clang on OS X 10.7 and later, but not with gcc on 10.6.

In short, the following works for me with clang on OS X > 10.6 with clang:
```sh
env LIBRARY_PATH='/opt/local/lib' /opt/local/bin/perl Configure.pl --prefix=/opt/local --has-dyncall
```
but it would be nice to respect `LDFLAGS`.

Here's a log from a [failed](https://build.macports.org/builders/ports-10.6_i386_legacy-builder/builds/13099/steps/install-port/logs/stdio) and [successful](https://build.macports.org/builders/ports-10.7_x86_64_legacy-builder/builds/19278/steps/install-port/logs/stdio) build.

**Comment by mojca** [comment]
@anddam managed to fix this by adding `$config->{ldflags}` to
```perl
$command = "$config->{ld} $config->{ldflags} $config->{ldout}$leaf $obj $config->{ldlibs} 2>&1";
```
in line 105 of [build/probe.pm](https://github.com/MoarVM/MoarVM/blob/90d7f1e86df9589769a2de910ebde3af49957895/build/probe.pm#L105), but I'm unable to reproduce that.

**Comment by MasterDuke17** [comment]
Does fe2cefb15e14865fdc5e7077647fff41e980b3e7 fix this issue?

**Comment by mojca** [comment]
Yes, thank you very much. I don't have 10.6 at hand, but our [buildbot](https://build.macports.org/builders/ports-10.6_x86_64_legacy-builder/builds/19933) is happy now.
